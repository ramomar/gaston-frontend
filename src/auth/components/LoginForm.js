import React from 'react';
import {
  Form,
  FormField,
  Button,
  Box,
  TextInput,
  Text
} from 'grommet';
import Spinner from '../../foundation/components/grommet/Spinner';

function logIn() {
  console.log('Loggin in...');
}

function LoginForm(props) {
  const initialValue = {};
  const hasError = false;

  return (
    <Form
      messages={{
        invalid: ':(',
        required: 'Requerido'
      }}
      value={initialValue}
      onSubmit={logIn}>
      <FormField
        required
        label='Usuario'
        name='user'
        component={TextInput}
        placeholder='yo@ramomar.dev' />
      <FormField
        required
        label='Contraseña'
        name='password'
        type='password'
        component={TextInput}
        placeholder='********' />
      <Box margin={{ top: 'large' }}>
        {hasError &&
          <Text textAlign='center' color='status-error'>
            ¡Ocurrió un error!
    </Text>}
      </Box>
      <Box>
        {(props.logInStatus ? props.loginStatus.isLoggingIn : false) ?
          <Box margin='small' pad='medium'>
            <Spinner />
          </Box> :
          <Button
            label={hasError ? 'Reintentar' : 'Continuar'}
            primary
            type='submit'
            margin='medium'
            alignSelf='center' />
        }
      </Box>
    </Form>
  );
}

// TODO: add props validations

export default LoginForm;
