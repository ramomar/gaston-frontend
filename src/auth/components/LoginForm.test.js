import React from 'react';
import renderer from 'react-test-renderer';
import { Grommet } from 'grommet';
import LoginForm from './LoginForm';


it('renders correctly', () => {
  const component =
    <Grommet>
      <LoginForm />
    </Grommet>;

  const tree = renderer.create(component);

  expect(tree).toMatchSnapshot();
});
