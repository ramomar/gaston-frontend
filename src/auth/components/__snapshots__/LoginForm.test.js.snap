// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk FormField__FormFieldBox-m9hood-0 eshLKp"
    >
      <label
        className="StyledText-sc-1sadyjn-0 bKLvLm"
      >
        Usuario
      </label>
      <div
        className="StyledBox-sc-13pk1d4-0 bDrWKZ"
      >
        <div
          className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
        >
          <input
            autoComplete="off"
            className="StyledTextInput-sc-1x30a0s-0 duvNfq"
            label="Usuario"
            name="user"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="yo@ramomar.dev"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk FormField__FormFieldBox-m9hood-0 eshLKp"
    >
      <label
        className="StyledText-sc-1sadyjn-0 bKLvLm"
      >
        Contraseña
      </label>
      <div
        className="StyledBox-sc-13pk1d4-0 bDrWKZ"
      >
        <div
          className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
        >
          <input
            autoComplete="off"
            className="StyledTextInput-sc-1x30a0s-0 duvNfq"
            label="Contraseña"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="********"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 hLaOSi"
    />
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <button
        className="StyledButton-sc-323bzc-0 cSMahv"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        Continuar
      </button>
    </div>
  </form>
</div>
`;
